src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseStringContainsCharOverloadWithSingleCharactersTests.cs

```diff
@@ -0,0 +1,168 @@
+}";
+            await VerifyCSCodeFixWithGivenReferenceAssemblies(violatingSourceCode, ReferenceAssemblies.NetStandard.NetStandard21, fixedSourceCode);
+        }
+
+        [Fact]
+        public async Task CSharp_ConstructedStringLiteral_Unchanged()
+        {
+            var sourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        return ""testString"".Contains(""a""+""samplesuffix"");
+    }
+}";
+            await VerifyCSCodeFixWithGivenReferenceAssemblies(sourceCode, ReferenceAssemblies.NetStandard.NetStandard21, sourceCode);
+        }
+
+        [Fact]
+        public async Task CSharp_ConstructedStringLiteralWithStringComparisonOverload_Unchanged()
+        {
+            var sourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        return ""testString"".Contains(""a""+""samplesuffix"", StringComparison.InvariantCultureIgnoreCase);
+    }
+}";
+            await VerifyCSCodeFixWithGivenReferenceAssemblies(sourceCode, ReferenceAssemblies.NetStandard.NetStandard21, sourceCode);
+        }
+
+        [Fact]
+        public async Task CSharp_SingleCharacterStringVariable_Unchanged()
+        {
+            var sourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        var singleCharacterString = ""a"";
+        return ""testString"".Contains(singleCharacterString);
+    }
+}";
+            await VerifyCSCodeFixWithGivenReferenceAssemblies(sourceCode, ReferenceAssemblies.NetStandard.NetStandard21, sourceCode);
+        }
+
+        [Fact]
+        public async Task CSharp_SingleCharacterStringVariableWithStringComparisonOverload_Unchanged()
+        {
+            var sourceCode = @"using System;
+public class TestClass
+{
+    public bool TestMethod(string input)
+    {
+        var singleCharacterString = ""a"";
+        return ""testString"".Contains(singleCharacterString, StringComparison.InvariantCultureIgnoreCase);
+    }
+}";
+            await VerifyCSCodeFixWithGivenReferenceAssemblies(sourceCode, ReferenceAssemblies.NetStandard.NetStandard21, sourceCode);
+        }
+
+        [Fact]
+        public async Task VB_RegularStringLiteralWithoutAvailableOverload_NoDiagnostic()
+        {
+            var test = new VerifyVB.Test()
+            {
+                TestCode = @"Public Class Program
+    Public Sub M()
+        Dim a = ""testString"".Contains(""a"")
+    End Sub
+End Class",
+                ReferenceAssemblies = ReferenceAssemblies.NetStandard.NetStandard20
+            };
+
+            await test.RunAsync();
+        }
+
+        [Fact]
+        public async Task VB_RegularStringLiteral_Fixed()
+        {
+            var test = new VerifyVB.Test()
+            {
+                TestCode = $@"Public Class Program
+    Public Sub M()
+        Dim a = ""testString"".Contains({{|{UseStringContainsCharOverloadWithSingleCharactersAnalyzer.CA1839}:""t""|}})
```

Comment:
For simplicity I think you can use `[|` `|]` instead of `{|CA1839:` `|}`. This will make things shorter and avoids the need of interpolation.

Reply:
Will the former notation default to the analyzer listed in the type variable or would this allow any diagnostic to fulfill the implicit assertion?
On another note i was looking for a documentation regarding the marker syntax but was ultimately unable so i just went off usage in the repo - is there anything to refer to?

Reply:
> Will the former notation default to the analyzer listed in the type variable or would this allow any diagnostic to fulfill the implicit assertion?

This allows only diagnostics from the analyzer under tested (i.e, compile errors for example won't be fulfilled). It also requires that the analyzer under test has only one rule id.

Tagging @sharwell for enhancing the documentation in [this README](https://github.com/dotnet/roslyn-sdk/blob/master/src/Microsoft.CodeAnalysis.Testing/README.md).